libmcount/dynamic.c

```diff
@@ -475,46 +475,57 @@ static void release_pattern_list(void)
 	}
 }
 
-static void patch_func_matched(struct mcount_dynamic_info *mdi,
-			       struct uftrace_mmap *map)
+static bool skip_sym(struct sym *sym, struct mcount_dynamic_info *mdi,
+		     struct uftrace_mmap *map)
 {
-	bool found = false;
-	struct symtab *symtab;
-	bool csu_skip;
-	unsigned i, k;
-	struct sym *sym;
 	/* skip special startup (csu) functions */
 	const char *csu_skip_syms[] = {
 		"_start",
 		"__libc_csu_init",
 		"__libc_csu_fini",
 	};
 	char *soname = get_soname(map->libname);
```

Comment:
Now, the soname is allocated and free'd for every symbol.

Reply:
right. I changed it again. thanks.